--- a/debian/rules
+++ b/debian/rules
@@ -1,6 +1,5 @@
 #!/usr/bin/make -f
 export PYBUILD_NAME=checkbox-support
-export SRCTOP=checkbox-support
 
 # DEBVER is the latest version as declared by the changelog
 #   it is in this form pkg_version~ubuntu(ubuntu_version)
@@ -8,8 +7,8 @@
 # This is not a valid python version, remove the ~ubuntu(ubuntu_version)
 # replace ~dev -> .dev as PEP440 mandates
 DEBVER := $(shell dpkg-parsechangelog -SVersion)
-PYTHON_VERSION := $(shell echo $(DEBVER) | sed 's/.ubuntu.\+//' | sed 's/~dev/.dev/')
+PYTHON_VERSION := $(shell echo $(DEBVER) | sed 's/.ubuntu.\+//' | sed 's/gl[0-9]//')
 export SETUPTOOLS_SCM_PRETEND_VERSION=$(PYTHON_VERSION)
 
 %:
-	dh $@ --sourcedirectory=$(SRCTOP) --with=python3 --buildsystem=pybuild
+	dh $@ --with=python3 --buildsystem=pybuild
